docs/src/ref_impl/build_and_run.md

```diff
@@ -159,5 +160,29 @@ In the case of i.MX8, Ghaf deployment contains of creating a bootable SD card wi
 2. To build and flash the Ghaf image:
    1. Run the `nix build .#packages.aarch64-linux.imx8qm-mek-release` command.
    2. Prepare the USB boot media with the target HW image you built: `dd if=./result/nixos.img of=/dev/<YOUR_USB_DRIVE> bs=32M`.
-   
+
 3. Insert an SD card and USB boot media into the board and switch the power on.
+
+---
+
+### Building Ghaf Image for Microchip Icicle kit
+
+Before you begin, check device-independent [prerequisites](./build_and_run.md#prerequisites).
+
+In the case of Icicle kit, Ghaf deployment contains of creating a SD Image which includes UBoot and Linux kernel from Microchip and Ghaf based NixOS rootfs.
+
+1. Build a Ghaf SD image:
```

Comment:
Replace with:

1. Build a Ghaf SD image:

   a. Run the `nix build .#packages.riscv64-linux.microchip-icicle-kit-release` command to release the image.
   b. Run the `nix build .#packages.riscv64-linux.microchip-icicle-kit-debug` command to debug the image.

Reply:
Done.